test/cqlpy/test_filtering.py

```diff
@@ -445,13 +470,13 @@ def test_multi_column_relation_tuples_null_check(cql, test_keyspace):
 #This test shows how Scylla handles invalid number of items in one of the tuples of multi-column relation.
 #If the number of items is lesser than required, Scylla throws "Expected {} elements in value tuple, but got {}"
 #But if the number of items is greater than required, Scylla throws different error: 
-#"Invalid list literal for in((b,c,d)): value (1, 2, 3, 4) is not of type frozen<tuple<int, int, int>>"
+#"Invalid list literal for IN((b,c,d)): value (1, 2, 3, 4) is not of type frozen<tuple<int, int, int>>"
 # Reproduces #13217
 def test_multi_column_relation_wrong_number_of_items_in_tuple(cql, test_keyspace):
     with new_test_table(cql, test_keyspace, 'a int, b int, c int, d int, primary key (a, b, c, d)') as table:
         with pytest.raises(InvalidRequest, match='elements in value tuple, but got'):
             cql.execute(f'SELECT * FROM {table} WHERE a = 0 AND (b, c, d) IN ((1, 2), (2, 1, 4))')
-        with pytest.raises(InvalidRequest, match='Invalid list literal for in'):
+        with pytest.raises(InvalidRequest, match='Invalid list literal for IN'):
```

Comment:
This is a pre-existing bug in this test, but it doesn't pass on Cassandra before or after your change. The error message is just different in Cassandra, it is "Invalid null value for b in tuple (b, c, d)". You can add both messages - if you think both are fine - using an "|" (the match option to raises() is a regular expression).

Reply:
I'll do it in a follow up, since it's not strictly related to NOT IN.

Also, note that Cassandra changes its error messages from time to time. So it could have worked in the past.